internals/plan/plan.go

```diff
@@ -275,23 +255,22 @@ func CommandString(base, extra []string) string {
 }
 
 // LogsTo returns true if the logs from s should be forwarded to target t.
-// This happens if:
-//   - t.Selection is "opt-out" or empty, and s.LogTargets is empty; or
-//   - t.Selection is not "disabled", and s.LogTargets contains t.
 func (s *Service) LogsTo(t *LogTarget) bool {
-	if t.Selection == DisabledSelection {
-		return false
-	}
-	if len(s.LogTargets) == 0 {
-		if t.Selection == UnsetSelection || t.Selection == OptOutSelection {
+	// Iterate backwards through t.Services until we find something matching
+	// s.Name.
```

Comment:
This is a really clever way to do it. I *think* I've convinced myself it has the same effect as keeping a set and evaluating left to right...

Reply:
`TestLogsTo` should remove any doubt